ruby/ql/src/queries/modeling/internal/Util.qll

```diff
@@ -0,0 +1,126 @@
+/**
+ * Contains utility methods and classes to assist with generating data extensions models.
+ */
+
+private import ruby
+private import codeql.ruby.ApiGraphs
+
+/**
+ * A file that is relevant in the context of library modeling.
+ *
+ * In practice, this means a file that is not part of test code.
+ */
+class RelevantFile extends File {
+  RelevantFile() { not this.getRelativePath().regexpMatch(".*/?test(case)?s?/.*") }
+}
+
+/**
+ * Gets an access path of an argument corresponding to the given `paramNode`.
+ */
+string getArgumentPath(DataFlow::ParameterNode paramNode) {
+  paramNode.getLocation().getFile() instanceof RelevantFile and
+  exists(Ast::Parameter param, string paramSpecifier |
+    param = paramNode.asParameter() and
+    (
+      paramSpecifier = param.getPosition().toString()
+      or
+      paramSpecifier = param.(Ast::KeywordParameter).getName() + ":"
+      or
+      param instanceof Ast::BlockParameter and
+      paramSpecifier = "block"
+    )
+  |
+    result = "Argument[" + paramSpecifier + "]"
+  )
```

Comment:
Should we include `self` arguments as well?
```ql
paramNode instanceof DataFlow::SelfParameterNode and paramSpecifier = "self"
```

Reply:
Added this